### PR TITLE
Backport script

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Rewrite cherry-pick/backport task to PyGitHub API

It runs for 30 seconds now instead of 1h40, see the comment below.

### CODE REVIEW

**Comment by Felixoid** [comment]
Annnnnnnd it's broken


https://github.com/ClickHouse/ClickHouse/runs/7356488540?check_suite_focus=true

**Comment by Felixoid** [comment]
Aaaaaand it works! https://github.com/ClickHouse/ClickHouse/runs/7357708663?check_suite_focus=true#step:5:1

**Comment by Felixoid** [comment]
Wow, it works! https://github.com/ClickHouse/ClickHouse/pull/39268

File: tests/ci/github_helper.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        """Wrapper around search method with throttling and splitting by date.
+
+        We split only by the first"""
+        splittable = False
+        for arg, value in kwargs.items():
+            if arg in ["closed", "created", "merged", "updated"]:
+                if (
+                    hasattr(value, "__iter__")
+                    and not isinstance(value, str)
+                    and not splittable
```
Comment (Felixoid) [inline tests/ci/github_helper.py:42]: oh no, it's wrong. The splittable should be checked later for preserved_arg, middle_value and splittable, but the `kwargs[arg] = f"{value[0].isoformat()}..{value[1].isoformat()}"` should be still done for every array or tuple in the kwargs

File: tests/ci/github_helper.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                    middle_value = value[0] + (value[1] - value[0]) / 2
+                    splittable = middle_value not in value
+                    kwargs[arg] = f"{value[0].isoformat()}..{value[1].isoformat()}"
+                    continue
+                assert isinstance(value, (date, datetime, str))
+
+        inter_result = []  # type: Issues
+        for i in range(3):
+            try:
+                logger.debug("Search issues, args=%s, kwards=%s", args, kwargs)
```
Comment (Felixoid) [inline tests/ci/github_helper.py:57]: kwards...

File: tests/ci/github_helper.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                return cached_pr
+        for i in range(3):
+            try:
+                pr = repo.get_pull(number)
+                break
+            except RateLimitExceededException:
+                if i == 2:
+                    raise
+                self.sleep_on_rate_limit()
+        logger.debug("Getting PR #%s from API", number)
```
Comment (Felixoid) [inline tests/ci/github_helper.py:147]: This line should go before loop

File: tests/ci/github_helper.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        def _get_pr(path: Path) -> PullRequest:
+            with open(path, "rb") as prfd:
+                return self.load(prfd)  # type: ignore
+
+        if pr_cache_file.is_file():
+            cached_pr = _get_pr(pr_cache_file)
+            if updated_at <= cached_pr.updated_at:
+                logger.debug("Getting PR #%s from cache", number)
+                return cached_pr
+        for i in range(3):
```
Comment (Felixoid) [inline tests/ci/github_helper.py:139]: this `3` should be a property

**Review by nickitat — APPROVED** [review]

**Comment by Felixoid** [comment]
Thank you for review, I'll merge it tomorrow morning at ~8-10 UTC

**Review by yakov-olkhovskiy — APPROVED** [review]

**Pull request merged**